src/Microsoft.ML.Core/CommandLine/CmdParser.cs

```diff
@@ -495,6 +500,11 @@ private static ArgumentInfo GetArgumentInfo(Type type, object defaults)
 
                     if (map.ContainsKey(name.ToLowerInvariant()))
                         throw Contracts.Except("Duplicate name '{0}' in argument type '{1}'", name, type.Name);
+                    if (attrName != null)
+                    {
+                        if (map.ContainsKey(attrName.ToLowerInvariant()))
```

Comment:
You may also construct the map with a case insensitive string compare, that way you dont have to do a lower variant. 

See here:  [https://stackoverflow.com/questions/13988643/case-insensitive-dictionary-with-string-key-type-in-c-sharp](https://stackoverflow.com/questions/13988643/case-insensitive-dictionary-with-string-key-type-in-c-sharp)